### ISSUE TITLE
adding automated dynamic speed control?

### ISSUE THREAD

**Issue by sharan21** [igrigorik/videospeed#548]
While watching study videos, there's a general pattern in that the introduction of the video is trivial(in understanding) and the middle section and later parts are little harder to digest and need more time. To make learning more effective it would make sense if the video speed was a dynamic function of time that could be set by the user so that the speed could be updated at regular intervals (say 5 seconds) and start fast and decrease speed gradually.  

It would be appreciated if this could be considered for future work.

**Comment by ChadBailey** [comment]
I follow the general concept, but how specifically would this be designed? What specific playback rates should be set and when should they be set?

**Comment by Eisys** [comment]
I'm guessing he means the video would start at 3x for example, and decrease in speed by perhaps 0.01x every x seconds. Or in a way that no matter how long the video is, it starts at (X)x and ends at 1x doing so gradually.

Interesting concept but seems pretty useless to most. If you're having trouble following something, just go back and watch/listen slower or just lower the speed yourself before you think a harder part is coming up.

**Comment by sharan21** [comment]
Yes, that's what I mean, it can be done manually but that may interrupt the
users viewing experience. Yes, it does seem useless to most, but I think it
may catch on if it is provided as an option, all new good ideas seem absurd
at first.

**Comment by ChadBailey** [comment]
Still unsure how I feel about this one... Anyone else want to weigh in with their thoughts?

**Comment by volotat** [comment]
I would really like this feature implemented. What I would like to use it for is to start viewing at some speed and then very gradually speed it up to train the brain to understand information at an even faster rate, but in such a way that it is noticeable for immediate attention.
